edx/edx-enterprise
# [ENT-348] Update course enrollment view

**Description:**

Building on the existing work, this pull request adds the capability to pull down actual course modes linked to the desired course and do the right thing when the submit button is pushed.

**JIRA:** [ENT-348](https://openedx.atlassian.net/browse/ENT-348) 

**Merge deadline:** 22 May 2017

**Sandbox link:** https://pr15025.sandbox.opencraft.hosting/enterprise/83f24af0-7077-489a-8a17-8ca4eafec6bb/course/course-v1%3AedX%2BDemoX%2BDemo_Course/enroll/

**Installation instructions:** Standard Enterprise setup

**Testing instructions:**

1. Fiddle with details of course (update course modes, logo, etc), and verify that all changes are reflected in the sandbox view. Note that updating course modes may take a few minutes to propagate; the backend REST response is cached.

2. Verify that selecting a premium mode kicks you out to the commerce flow (the standard edX shopping cart on this sandbox) and that selecting the audit mode takes you directly to the course, with consent if necessary.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
